src/main/java/com/direwolf20/buildinggadgets/common/schema/BGTemplate.java

```diff
@@ -0,0 +1,68 @@
+package com.direwolf20.buildinggadgets.common.schema;
+
+import com.google.common.collect.AbstractIterator;
+import it.unimi.dsi.fastutil.longs.Long2ObjectMap;
+import it.unimi.dsi.fastutil.longs.Long2ObjectMap.Entry;
+import it.unimi.dsi.fastutil.objects.ObjectIterator;
+import net.minecraft.block.BlockState;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.math.BlockPos.Mutable;
+import net.minecraft.util.math.Vec3i;
+
+import java.util.Iterator;
+
+public final class BGTemplate implements Iterable<MutableBlockData> {
```

Comment:
Not a fan on the name at all, either Template or Schema

Reply:
Template is easy to confuse with MC's Template class, however the Term schema does not really specify what it is (it is not a schematic image from it, which may leave out details to clarify the whole, but rathe a Template which serves as a model for building stuff of the sort), therefore I'd go for Template then.